src/vma/dev/ring.h

```diff
@@ -271,24 +278,27 @@ class ring : public mem_buf_desc_owner
 		m_lock_ec_list.unlock();
 	}
 
+	inline void del_ec(struct ring_ec *ec)
+	{
+		m_lock_ec_list.lock();
+		list_del_init(&ec->list);
+		ec->clear();
+		m_lock_ec_list.unlock();
+	}
+
 	inline ring_ec* get_ec(void)
 	{
 		struct ring_ec *ec = NULL;
 
 		m_lock_ec_list.lock();
 		if (!list_empty(&m_ec_list)) {
 			ec = list_entry(m_ec_list.next, struct ring_ec, list);
-			list_del(&ec->list);
+			list_del_init(&ec->list);
```

Comment:
Igor previously there was delete of one element from the list, now it looks you are dropping after the whole list?